__tests__/components/cards/challenge/_partials/ReplyToInvitation.test.tsx

```diff
@@ -0,0 +1,79 @@
+import { render, screen, fireEvent, waitFor } from "@testing-library/react";
+import "@testing-library/jest-dom";
+import { useDispatch } from "@/hooks/useTypedDispatch";
+import { useMultiSelector } from "@/hooks/useTypedSelector";
+import ReplyToInvitation, { InvitationProps } from "@/components/cards/challenge/_partials/ReplyToInvitation";
+import { acceptInvitation, declineInvitation } from "@/store/feature/communities/challenges/invites.slice";
+jest.mock("@/hooks/useTypedDispatch");
+jest.mock("@/hooks/useTypedSelector");
+jest.mock("@/store/services/teams.service");
+jest.mock("@/store/feature/communities/challenges/invites.slice");
+const mockDispatch = jest.fn();
+
+(useDispatch as jest.Mock).mockReturnValue(mockDispatch);
+(useMultiSelector as jest.Mock).mockReturnValue({
+  challenge: { id: "challenge-id" },
+  team: { invites: [{ id: "invite-id", status: "PENDING" }] },
+});
+
+const invitationProps: InvitationProps = {
+  invite_id: "invite-id",
+  team_ref: "team/1",
+  ReplyToInvitationTestId: "reply-to-invitation"
+};
+
+describe("ReplyToInvitation", () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it("should render the ReplyToInvitation component", () => {
+    render(<ReplyToInvitation {...invitationProps} />);
+    expect(screen.getByTestId(invitationProps.ReplyToInvitationTestId!)).toBeInTheDocument();
+  });
+
+  it("should render accept button and decline button in component",() => {
+    render(<ReplyToInvitation {...invitationProps} />);
+    const acceptButton = screen.getByTestId(invitationProps.ReplyToInvitationTestId!);
+    const declineButton = screen.getByTestId(invitationProps.ReplyToInvitationTestId!);
+    expect(acceptButton).toBeInTheDocument();
+    expect(declineButton).toBeInTheDocument();
+  });
+
+  it("should display loader when loading", async () => {
+    render(<ReplyToInvitation {...invitationProps} />);
+    expect(screen.getByTestId("loader")).toBeInTheDocument();
+    await waitFor(() => expect(screen.queryByTestId("loader")).not.toBeInTheDocument());
```

Comment:
to improve the code you should use `findByTestId` instead if `waitFor` since it is more straightforward approach and in case of an error  `findByTestId` provide clear error messages that helps to debug 
```suggestion
 expect(await screen.findByTestId('loader')).not.toBeInTheDocument()
```